OpenMTRDemo/Forms/TiledFiltersForm.cs

```diff
@@ -37,77 +39,72 @@ public override void loadToolStripMenuItem_Click(object sender, EventArgs e)
             LoadSaveDialog loadSaveDialog = new LoadSaveDialog();
             if (loadSaveDialog.openBrowser.ShowDialog() == DialogResult.OK)
             {
-                _meter = new MeterImage(loadSaveDialog.openBrowser.FileName, new Mat(loadSaveDialog.openBrowser.FileName), new Mat(loadSaveDialog.openBrowser.FileName));
+                _meter = new MeterImage(loadSaveDialog.openBrowser.FileName, new Mat(loadSaveDialog.openBrowser.FileName));
                 LoadAllImagePanes();
             }
         }
 
         private void LoadAllImagePanes()
         {
-            _cannyMat = _meter.SourceImage.Clone();
-            _grayMat = _meter.SourceImage.Clone();
-            _sobelMat = _meter.SourceImage.Clone();
-            _laplacianMat = _meter.SourceImage.Clone();
-            _scharrMat = _meter.SourceImage.Clone();
+            _cannyMat = _meter.Clone();
+            _grayMat = _meter.Clone();
+            _sobelMat = _meter.Clone();
+            _laplacianMat = _meter.Clone();
+            _scharrMat = _meter.Clone();
+            _meterPane1 = _meter.Clone();
+            _meterPane2 = _meter.Clone();
+            _meterPane3 = _meter.Clone();
+            _meterPane4 = _meter.Clone();
+            _meterPane5 = _meter.Clone();
+            _meterPane6 = _meter.Clone();
 
-            _meterPane1 = new MeterImage();
-            _meterPane2 = new MeterImage();
-            _meterPane3 = new MeterImage();
-            _meterPane4 = new MeterImage();
-            _meterPane5 = new MeterImage();
-            _meterPane6 = new MeterImage();
+            _cannyMat.Add(new CannyFilter());
+            _grayMat.Add(new GrayFilter());
+            _sobelMat.Add(new SobelFilter());
+            _laplacianMat.Add(new LaplacianFilter());
+            _scharrMat.Add(new ScharrFilter());
 
-            _meterPane1.SourceImage = _meter.SourceImage.Clone();
-            _meterPane2.SourceImage = _meter.SourceImage.Clone();
-            _meterPane3.SourceImage = _meter.SourceImage.Clone();
-            _meterPane4.SourceImage = _meter.SourceImage.Clone();
-            _meterPane5.SourceImage = _meter.SourceImage.Clone();
-            _meterPane6.SourceImage = _meter.SourceImage.Clone();
+            _meterList = new List<KeyValPair<PictureBox>>();
 
-            //OpenCVSharp required Mat data type to not be null when databound.
-            _meterPane1.ModifiedImage = _meter.SourceImage.Clone();
-            _meterPane2.ModifiedImage = _meter.SourceImage.Clone();
-            _meterPane3.ModifiedImage = _meter.SourceImage.Clone();
-            _meterPane4.ModifiedImage = _meter.SourceImage.Clone();
-            _meterPane5.ModifiedImage = _meter.SourceImage.Clone();
-            _meterPane6.ModifiedImage = _meter.SourceImage.Clone();
+            _meterList.Add(new KeyValPair<PictureBox>(pictureBoxSource ,_meter));
```

Comment:
I would probably load the list of key value pairs in a different method than 'LoadAllImagePanes'

Reply:
Should I put that into it's own method, then?